[Program: GCI] Add missing tests for Change Password API

ilyamaier opened the pull request:
### Description
- Added missing tests for Change Password API
- Added `addUser()` function to avoid repeating code

Fixes #308 

### Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
This was tested on my laptop.

### Screenshot:
![Screenshot (44)](https://user-images.githubusercontent.com/34242059/71424728-42262480-2695-11ea-8d34-1a903a8bf10d.png)

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

Commit:
test: Add missing tests for Change Password API

test: Add missing tests for Change Password API

Tlazypanda commented on tests/users/test_api_update_user.py:
```diff
@@ -165,15 +146,103 @@ def test_update_availability_to_be_mentee_to_false(self):
+        self.assertDictEqual(expected_response,
+                             json.loads(actual_response.data))
+
+    def test_change_password_to_current_password(self):
```
@IlyaMaier Can you remove this test as discussed?

Tlazypanda commented on tests/users/test_api_update_user.py:
```diff
@@ -37,121 +32,107 @@ def test_update_username_already_taken(self):
         db.session.add(self.first_user)
         db.session.commit()
 
+    def test_update_user_api_resource_non_auth(self):
```
Can you just add only the tests and leave the pep8 issues and other code changes for now? In opensource it is considered a good practice to only fix one issue per pull request.Thanks :)

Tlazypanda reviewed (changes requested):
@IlyaMaier Amazing work! :fire: Just a couple of more changes and we are good to go!

ilyamaier commented:
> @IlyaMaier Amazing work! 🔥 Just a couple of more changes and we are good to go!

@Tlazypanda thank you for a review! I deleted all irrelevant stuff and updated the PR's message.

Tlazypanda commented on app/messages.py:
```diff
@@ -135,6 +135,11 @@
 USER_ENTERED_INCORRECT_PASSWORD = {"message": "Current password is incorrect."}
+USER_ENTERED_CURRENT_PASSWORD = {"message": "New password should not be same "
+                                            "with the current password."}
+WRONG_LENGTH_OF_PASSWORD = {"message": "The new_password field has to be "
```
Can you check this method `test_user_password_change_request_password_inferior_to_limit` and implement the message like this instead of adding a new one?

Tlazypanda commented on tests/users/test_api_update_user.py:
```diff
@@ -174,6 +188,83 @@ def test_update_availability_to_be_mentee_to_false(self):
+    def test_change_password_without_token(self):
+        self.add_user()
+        auth_header = {'Authorization': 'Bearer {}'.format("")}
+        expected_response = messages.TOKEN_IS_INVALID
```
Here auth token should be missing and message should be `AUTHORISATION_TOKEN_IS_MISSING`

Tlazypanda commented on app/messages.py:
```diff
@@ -135,6 +135,11 @@
 
 # Login errors
 USER_ENTERED_INCORRECT_PASSWORD = {"message": "Current password is incorrect."}
+USER_ENTERED_CURRENT_PASSWORD = {"message": "New password should not be same "
```
Can you remove this as well since it is not invoked anymore?

Tlazypanda reviewed (changes requested):
@IlyaMaier Good job! :clap: Requested a few more changes on the pr :)

Tlazypanda reviewed (approved):
@IlyaMaier Approved :)

isabelcosta commented:
Thank you for your contribution @IlyaMaier  🎉 
I'm closing this PR because we decided to merge another GCI PR.

ilyamaier commented:
> Thank you for your contribution @IlyaMaier 
> I'm closing this PR because we decided to merge another GCI PR.

Okay! Thanks for letting me know!
